ui/views/tx/tx-view.js

```diff
@@ -19,7 +19,20 @@ export default withErrorBoundary(function TxView() {
     const [txInfo, setTxInfo] = useDependantState(() => {
         setError('')
         loadTx(txhash)
-            .then(txInfo => setTxInfo(txInfo))
+            .then(async txInfo => {
+                const horizonTx = await checkTxSubmitted(txInfo)
+                if (!horizonTx.submitted)
+                    return setTxInfo(txInfo)
+                //Send transaction to the server
+                await submitTx({
```

Comment:
?
We should submit tx only if auto-submit is enabled. And we do it on the server side.
I think this logic is wrong.